Add object type filtering to Social Web feed sidebar

pfefferle opened the pull request:
## Summary
Adds object type filtering functionality to the Social Web feed sidebar, allowing users to filter feed posts by ActivityPub object types (Articles, Notes, Videos, etc.).

<img width="2020" height="1183" alt="Screenshot 2025-11-25 at 11 22 25" src="https://github.com/user-attachments/assets/a8cf25f1-d9a1-40d7-979f-34eca0b7715b" />

(the list should normally way smaller and shows only the known types that were imported at least once)

## Changes
- Created `ObjectTypes` component that displays taxonomy terms as clickable filter items
- Created `useObjectTypeFilter` hook to manage filter state using DataViews
- Integrated component into sidebar navigation below the Feed menu item
- Added icons for each object type (Articles, Music & Podcasts, Videos, etc.)
- Added postList icon to Feed menu item
- Configured object type field with 'is' operator for proper DataViews integration

## Implementation Details
- Object types are fetched from `ap_object_type` taxonomy
- Filter uses single selection with toggle behavior (click to filter, click again to clear)
- Selected type is highlighted in the sidebar
- Filter button in DataViews toolbar displays the selected type name
- Values use translated display names matching the existing filter implementation
- Icons are 24x24 for consistency

## Test Plan
- [ ] Click object type in sidebar filters feed correctly
- [ ] Selected type is highlighted in sidebar
- [ ] Type filter button in toolbar shows correct label
- [ ] Click same type again clears the filter
- [ ] Switching between types updates both sidebar and feed

Fixes #2510

pfefferle committed:
Add object type filtering to feed sidebar with icons.

- Create useObjectTypeFilter hook to manage ap_object_type filters
- Add ObjectTypes component to display taxonomy terms in sidebar
- Integrate ObjectTypes under Feed menu item using MenuItem design
- Add postList icon to Feed menu item
- Add icons for each object type (article, audio, document, event, etc.)
- Add default icon (post) for unmapped types
- Enable toggle filtering by clicking object type names
- Fix selection highlight by manually adding is-selected class
- Use simple margin-right (8px) for icon spacing matching Gutenberg
- Use same translated names as built-in object type field
- Follow same pattern as tag filtering using view.filters

pfefferle committed:
Fix object type filter to use 'is' operator with proper value handling

- Keep object type field filter operator as 'is' (not 'isAny')
- Update hook to use single number value instead of array
- Update useFeed hook to wrap single values in array for REST API compatibility
- Update icon sizes from 20 to 24 for both Feed menu item and object types
- This fixes the issue where the "Type" filter button in DataViews toolbar
  doesn't properly display the selected object type name

pfefferle committed:
Merge branch 'add/reader' into add/object-type-filter

Integrate with popular tags feature from PR #2512

pfefferle committed:
Keep PopularTags in its original position outside nav element

pfefferle committed:
Use aria-pressed and add loading state for WordPress consistency

- Replace isSelected prop with aria-pressed attribute following WordPress accessibility patterns
- Remove manual .is-selected class addition
- Add loading state with proper wrapper div and message
- Update CSS to handle aria-pressed='true' for selection state
- Add styling for loading state with BEM naming

pfefferle committed:
Fix focus and active states to prevent blue glow and background flashing

- Remove default focus box-shadow and outline
- Add focus-visible for keyboard navigation only
- Use !important to override MenuItem default styles
- Keep background consistent on active/hover for selected items
- Matches WordPress core behavior

pfefferle committed:
Make Feed menu item selected when no filters active and clear filters on click

- Created useFeedFilters hook to detect active filters and clear all filters
- Feed menu item now shows as selected when no filters are active
- Clicking Feed clears all active filters (object type and tag filters)
- Disabled transitions on menu items to prevent glow/blinking effects
- Added !important to all focus/hover/active states for consistent behavior

pfefferle committed:
Use file icon for Document object type instead of media icon

pfefferle committed:
Use help icon as default for unknown object types

pfefferle committed:
Fix object type field to fetch all terms matching sidebar behavior

The field definition was using default WordPress query parameters which limited results to 10 items. Now uses the same parameters as the sidebar component (per_page: -1, orderby: count, order: desc, hide_empty: true) to ensure all object types appear in the filter dropdown.

pfefferle committed:
Show only known object types in sidebar

Filter sidebar to only display object types that have translations (Article, Audio, Document, Event, Image, Note, Page, Place, Video). Unknown types like Collection, Question, Activity, and Offer will still appear in the filter dropdown but not in the sidebar navigation.

pfefferle commented:
@obenland thanks for cleaning up the tags pr #2512 !

I tried to adapt the hooks as best as possible and brought back the icons, to better match the site-editor look and feel.

obenland commented on src/social-web/components/fields/object-type/index.tsx:
```diff
@@ -25,7 +25,12 @@ export const objectTypeField: Field< FeedPost > = {
-		const records: Term[] = await resolveSelect( coreDataStore ).getEntityRecords( 'taxonomy', 'ap_object_type' );
+		const records: Term[] = await resolveSelect( coreDataStore ).getEntityRecords( 'taxonomy', 'ap_object_type', {
+			per_page: -1,
+			orderby: 'count',
```
Ordering by count could lead to the menu order changing, based on incoming activity and ap_post purges. I wonder if there's a way to maintain a specific order?

pfefferle replied:
good point!

obenland commented on src/social-web/components/object-types/index.tsx:
```diff
@@ -0,0 +1,100 @@
+	Page: __( 'Pages', 'activitypub' ),
+	Place: __( 'Places & Locations', 'activitypub' ),
+	Video: __( 'Videos', 'activitypub' ),
+};
```
Is this something we could export, so it can be re-used in the object-type field?

obenland commented on src/social-web/components/object-types/index.tsx:
```diff
@@ -0,0 +1,100 @@
+};
+
+// Default icon for unmapped types
+const defaultIcon = help;
```
```suggestion
```
It's unused.

pfefferle replied:
Oh, I had it in the menu before... good catch!

obenland commented on src/social-web/components/object-types/index.tsx:
```diff
@@ -0,0 +1,100 @@
+import { Icon, MenuItem, MenuGroup } from '@wordpress/components';
+import { sprintf, __ } from '@wordpress/i18n';
+import { useObjectTypeFilter } from '../../hooks/use-object-type-filter';
+import { postContent, audio, file, calendar, image, comment, page, pin, video, help } from '@wordpress/icons';
```
I get a TS error for `page`, saying `@wordpress/icons` doesn't have an export of that name.

pfefferle replied:
I will check!

pfefferle replied:
there is a page: https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library

obenland reviewed (commented):
Nice work!

pfefferle committed:
Sort object types by defined translation order

The object types menu now displays types in the order defined by the translations object, rather than by count or default taxonomy order. Unused icons and parameters were removed for clarity.

pfefferle committed:
Refactor object type labels and icons mapping

Moved object type label and icon mapping to a shared module and updated usage in feed-stage and object-type components. This improves maintainability and consistency for object type display throughout the social web UI.

pfefferle committed:
Merge branch 'add/reader' into add/object-type-filter

pfefferle committed:
Merge add/reader into add/object-type-filter

Resolves build file conflicts by regenerating after source merge.

Includes changes from add/reader:
- Content field improvements
- Object type context
- Test cleanup changes
- Updated hashtag handling in Posts collection
